src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,261 @@
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.serialization.StringDeserializer;
-import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.listener.ConsumerSeekAware;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.MessageHeaders;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.kafka.consumer.ConsumerConfig;
-import uk.gov.companieshouse.kafka.consumer.factory.KafkaConsumerFactory;
-import uk.gov.companieshouse.kafka.consumer.resilience.CHConsumerType;
-import uk.gov.companieshouse.kafka.consumer.resilience.CHKafkaResilientConsumerGroup;
-import uk.gov.companieshouse.kafka.exceptions.ProducerConfigException;
+import uk.gov.companieshouse.itemhandler.exception.RetryableErrorException;
 import uk.gov.companieshouse.kafka.exceptions.SerializationException;
 import uk.gov.companieshouse.kafka.message.Message;
-import uk.gov.companieshouse.kafka.producer.Acks;
-import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
-import uk.gov.companieshouse.kafka.producer.ProducerConfig;
 import uk.gov.companieshouse.kafka.serialization.AvroSerializer;
 import uk.gov.companieshouse.kafka.serialization.SerializerFactory;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 import uk.gov.companieshouse.orders.OrderReceived;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.ExecutionException;
 
 import static uk.gov.companieshouse.itemhandler.ItemHandlerApplication.APPLICATION_NAMESPACE;
 
 @Service
-public class OrdersKafkaConsumer implements InitializingBean {
+public class OrdersKafkaConsumer implements ConsumerSeekAware {
     private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
     private static final String ORDER_RECEIVED_TOPIC = "order-received";
     private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_KEY_RETRY = ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
     private static final String ORDER_RECEIVED_GROUP = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC;
     private static final String ORDER_RECEIVED_GROUP_RETRY = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_GROUP_ERROR = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_ERROR;
-    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static long errorRecoveryOffset = 0l;
```

Comment:
You appear to have removed the constant `MAX_RETRY_ATTEMPTS`. How is the number of retries set/configured/controlled now?

Reply:
This will be implemented with the help of [@Retryable](https://github.com/spring-projects/spring-retry) by the class that implements the retryable processing logic. The default number of retries of a `@Retryable` annotated method is 3, which made this constant redundant.

Reply:
So this changes the nature of the contract. Previously the assumption was that the consumer logic would determine for example how many retries should be attempted. Now it will be dictated by parameters  provided to a `@Retryable` annotation on the service method that is called by this consumer? I understand this may be the norm when using Spring Retry, but perhaps we need more explicit documentation somewhere here outlining the expectations of such a service?

Reply:
This has now changed as the resilience logic no longer relies on `@Retryable`.